src/object.js

```diff
@@ -128,10 +128,16 @@ inherits(ObjectSchema, MixedSchema, {
     let originalValue =
       opts.originalValue != null ? opts.originalValue : _value;
 
+    let fromClosure = (scope, from) => () => {
+      return { value: originalValue, schema: scope, from };
+    };
```

Comment:
this might be a bit simpler if instead of a nested closure, ancestors were stored in an array? e.g. `this.ancestors[0]`

Reply:
> this might be a bit simpler if instead of a nested closure, ancestors were stored in an array? e.g. `this.ancestors[0]`

I thought about that as well as an option, but I felt an API of
```
   const parent = from[0];
   const grandparent = from[1].value;
```

to feel less intuitive of:
```
   const parent = from();
   const grandparent = from().from();
```

But that's very much personal preference. It's your library, and if you like the concept, I'd be happy to update the PR!

Are you still okay with accessing the parentSchemas so it would look like:

```
const { value, schema } = from[0];
const { value: grandparentValue, schema: grandparentSchema } = from[1];
```